### PR TITLE
[Debug] Fix regex type names in DebugDescriptionMacro

### PR DESCRIPTION
**Pull request by kastiglione**
Fixes the way `DebugDescriptionMacro` produces a regex type name.

The problem was use of backslash escapes that weren't sufficiently escaped. They needed to be double escaped. To avoid this trap, the regexes now use `[.]` to match a dot, instead of the more conventional `\.` syntax.

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci smoke test

File: lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift [hunk 1]
```diff
@@ -71,15 +71,19 @@ extension DebugDescriptionMacro: MemberAttributeMacro {
       return []
     }
 
+    // Warning: To use a backslash escape in `typeIdentifier`, it needs to be double escaped. This is because
+    // the string is serialized to a String literal (an argument to `@_DebugDescriptionProperty`), which
+    // effectively "consumes" one level of escaping. To avoid mistakes, dots are matched with `[.]` instead
```
[the remark this answers is not in the document]
Comment (adrian-prantl) [reply lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift:76]: Apparently.
```
echo 'Hello. World.' | sed -e 's/[.]/!/g'
Hello! World!
```

**Review by adrian-prantl — APPROVED** [review]

File: lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (kastiglione) [reply lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift:76]: Yes, character classes match their contents literally (generally speaking, there could be some exceptions I am unaware of). This works for other metacharacters too, ex `[+]`, `[$]`, etc.

[the remark this answers is not in the document]
Comment (kastiglione) [reply lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift:76]: This kind of usage is used as an alternative to backslash escaping.

**Pull request merged**